src/teleop/gui/src/components/AutonWaypointEditor.vue

```diff
@@ -275,42 +428,48 @@ export default {
       this.input.lon.s = 0;
       this.input.lat = convertDMS(this.input.lat, this.odom_format_in);
       this.input.lon = convertDMS(this.input.lon, this.odom_format_in);
-    },
+    }
   },
+
   beforeDestroy: function () {
     window.clearInterval(interval);
+    this.autonEnabled = false;
+    this.sendEnableAuton();
   },
 
   created: function () {
     (this.course_pub = new ROSLIB.Service({
       ros: this.$ros,
       name: "/enable_auton",
-      serviceType: "mrover/PublishEnableAuton",
+      serviceType: "mrover/PublishEnableAuton"
     })),
       (this.nav_status_sub = new ROSLIB.Topic({
         ros: this.$ros,
         name: "/smach/container_status",
-        messageType: "smach_msgs/SmachContainerStatus",
+        messageType: "smach_msgs/SmachContainerStatus"
       })),
       (this.rover_stuck_pub = new ROSLIB.Topic({
         ros: this.$ros,
         name: "/rover_stuck",
-        messageType: "std_msgs/Bool",
+        messageType: "std_msgs/Bool"
       })),
-      this.nav_status_sub.subscribe(
-        (msg) => {
-          if (msg.active_states[0] !== "OffState" && !this.autonEnabled) {
-            return;
-          }
-          this.waitingForNav = false;
-          this.autonButtonColor = this.autonEnabled ? "green" : "red";
-        },
+      // Make sure local odom format matches vuex odom format
+      (this.odom_format_in = this.odom_format);
+
+    this.nav_status_sub.subscribe(
+      (msg) => {
+        if (msg.active_states[0] !== "OffState" && !this.autonEnabled) {
+          return;
+        }
+        this.waitingForNav = false;
+        this.autonButtonColor = this.autonEnabled ? "green" : "red";
+      },
 
-        // Interval for publishing Course
-        (interval = window.setInterval(() => {
-          this.rover_stuck_pub.publish({ data: this.roverStuck });
-        }, 100))
-      );
+      // Interval for publishing Course
+      (interval = window.setInterval(() => {
+        this.rover_stuck_pub.publish({ data: this.roverStuck });
+      }, 100))
```

Comment:
I'm not sure this is wrong, but I'm kinda confused what's happening here. What's the reasoning behind creating this interval inside of this subscribe thing?